lib/psabpf_table.c

```diff
@@ -497,6 +567,16 @@ int psabpf_matchkey_prefix(psabpf_match_key_t *mk, uint32_t prefix)
     return NO_ERROR;
 }
 
+/* only for lpm */
+uint32_t psabpf_matchkey_get_prefix(psabpf_match_key_t *mk)
+{
+    if (mk == NULL)
+        return 0;
+    if (mk->type != PSABPF_LPM)
```

Comment:
The comment above suggests that this method is for LPM but here I see non-LPM code ...

Reply:
This check is for case when someone calls this function on non-LPM key. Something must be returned, I assumed to behave as an exact match in such case.

Reply:
I see that `psabp_matchkey_get_prefix` is used only on LPM key. So taking into account non-LPM matches here is like "over programming". I strongly suggest to return 0 in non-LPM cases. This will tell a programmer to not use `...get_prefix_len` on non-LPM matches.

Reply:
Ok, it makes sense for me. But anyway, `0` is a valid prefix length...

Reply:
Done

Reply:
Yes, but I don't see any other possibilty. For me it's fine.